### PR TITLE
Compactness for classical propositional logic is a constructive taboo

### PR DESCRIPTION
**Pull request by TOTBWF**
# Description

This PR proves that compactness for classical propositional logic is a constructive taboo. It also defines weak excluded middle, and adds some helpers for working with types with cardinality 0.

### CODE REVIEW

**Comment by Lavenza** [comment]
[Pull request preview](https://preview.1lab.dev/341)

File: src/1Lab/Classical.lagda.md [hunk 1]
```diff
@@ -60,6 +61,35 @@ LEM≃DNE : LEM ≃ DNE
...
 ```
 
+## Weak excluded middle {defines="weak-excluded-middle"}
+
+The **weak law of excluded middle** (WLEM) is a slightly weaker variant
+of excluded middle that introduces an extra double negation.
+
+```agda
+WLEM : Type
+WLEM = ∀ (P : Ω) → ¬ ¬ ∣ P ∣ ⊎ ¬ ∣ P ∣
```
Comment (ncfavier) [inline src/1Lab/Classical.lagda.md:71]: `Dec (¬ ∣ P ∣)`, for consistency?

File: src/Logic/Propositional/Classical/Compactness.lagda.md [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+  [x∣P] : Proposition 1 → Ω
+  [x∣P] ϕ = elΩ ((x ≡ ϕ) × ∣ P ∣)
+
+  [¬x∣¬P] : Proposition 1 → Ω
+  [¬x∣¬P] ϕ = elΩ ((“¬” x ≡ ϕ) × (¬ ∣ P ∣))
+```
+
+There are 3 possible finite subsets of
+$\{ x \mid P \} \cup \{ \lnot x \mid \lnot P \}$.
+* $\emptyset$
```
Comment (ncfavier) [inline src/Logic/Propositional/Classical/Compactness.lagda.md:76]:
```suggestion

* $\emptyset$
```

**Comment by TOTBWF** [comment]
Fixed a merge conflict with the `bib` file, should be good to go!

**Review by ncfavier — APPROVED** [review]

File: src/Logic/Propositional/Classical/Compactness.lagda.md [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+There are 3 possible finite subsets of
+$\{ x \mid P \} \cup \{ \lnot x \mid \lnot P \}$.
+
+* $\emptyset$
+* $\{ x \}$
+* $\{ \lnot x \}$
+
+All three of these are clearly satisfiable, but
+"Unfortunately, proof assistants": convincing Agda that those are the only
+3 possible subsets involves quite a work.
```
Comment (plt-amy) [inline src/Logic/Propositional/Classical/Compactness.lagda.md:83]:
```suggestion
3 possible subsets involves quite a bit of work.
```
This paragraph needs to be re-line-wrapped

**Review by ncfavier — APPROVED** [review]

**Review by plt-amy — APPROVED** [review]

**Pull request merged**